modules/dists/StencilDist.chpl

```diff
@@ -784,6 +857,133 @@ iter StencilImpl.activeTargetLocales(const space : domain = boundingBox) {
+proc type Stencil.createArray(
+  dom: domain,
+  type eltType,
+  initExpr: ?t,
+  targetLocales: [] locale = Locales,
+  fluff = makeZero(dom.rank, dom.idxType),
+  periodic = false
+) where isSubtype(t, _iteratorRecord) || isCoercible(t, eltType)
+{
+  var D = createDomain(dom, targetLocales, fluff, periodic);
+  var A: [D] eltType;
+  A = initExpr;
+  return A;
+}
+
+// create an array over a Stencil Distribution, initialized from the given array
+proc type Stencil.createArray(
+  dom: domain,
+  type eltType,
+  initExpr: [?arrayDom] ?arrayEltType,
+  targetLocales: [] locale = Locales,
+  fluff = makeZero(dom.rank, dom.idxType),
+  periodic = false
+) where dom.rank == arrayDom.rank && isCoercible(arrayEltType, eltType)
+{
+  for (d, ad, i) in zip(dom.dims(), arrayDom.dims(), 0..) do
+    if d.size != ad.size then halt("Domain size mismatch in 'Stencil.createArray' dimension " + i:string);
+  var D = createDomain(dom, targetLocales, fluff, periodic);
+  var A: [D] eltType;
+  A = initExpr;
+  return A;
+}
+
+// create an array over a Stencil Distribution constructed from a series of ranges, default initialized
+proc type Stencil.createArray(
+  rng: range...?k,
+  type eltType,
+  targetLocales: [] locale = Locales,
+  fluff: ?f = makeZero(k, int),
+  periodic = false
+) where isHomogeneousTupleType(f) {
+  return createArray({(...rng)}, eltType, targetLocales, fluff, periodic);
+}
+
+proc type Stencil.createArray(rng: range...?k, type eltType) {
+  return createArray({(...rng)}, eltType, fluff = makeZero(k, rng[0].idxType));
+}
+
+// create an array over a Stencil Distribution constructed from a series of ranges, initialized with the given value or iterator
+proc type Stencil.createArray(rng: range...?k, type eltType, initExpr: ?t)
+  where isSubtype(t, _iteratorRecord) || isCoercible(t, eltType)
+{
+  return createArray({(...rng)}, eltType, initExpr, fluff = makeZero(k, rng[0].idxType));
+}
+
+proc type Stencil.createArray(
+  rng: range...?k,
+  type eltType,
+  initExpr: ?t,
+  targetLocales: [] locale = Locales,
+  fluff: ?f = makeZero(k, int),
+  periodic = false
+) where (isSubtype(t, _iteratorRecord) || isCoercible(t, eltType)) && isHomogeneousTupleType(f)  {
+  return createArray({(...rng)}, eltType, initExpr, targetLocales, fluff, periodic);
+}
+
+// create an array over a Cyclic Distribution constructed from a series of ranges, initialized from the given array
+proc type Stencil.createArray(
+  rng: range...?k,
+  type eltType,
+  initExpr: [?arrayDom] ?arrayEltType
+) where k == arrayDom.rank && isCoercible(arrayEltType, eltType)
+{
+  return createArray({(...rng)}, eltType, initExpr);
+}
+
+proc type Stencil.createArray(
+  rng: range...?k,
+  type eltType,
+  initExpr: [?arrayDom] ?arrayEltType,
+  targetLocales: [] locale = Locales,
+  fluff: ?f = makeZero(k, int),
+  periodic = false
+) where k == arrayDom.rank && isCoercible(arrayEltType, eltType) && isHomogeneousTupleType(f)
```

Comment:
One of the other cases where I'd rather do `rng.size == arrayDom.rank` rather than introducing `?k`.